ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/ContractsReader.scala

```diff
@@ -239,41 +328,48 @@ private[appendonlydao] sealed class ContractsReader(
     AND NOT EXISTS       -- check no archival visible
          (SELECT 1
             FROM participant_events, parameters
-           WHERE event_kind = 20 -- consuming exercise
+           WHERE event_kind = #$EventKindArchived
              AND event_sequential_id <= parameters.ledger_end_sequential_id
              AND #${flat_event_witnessesWhere("participant_events")}
              AND contract_id = last_contract_key_create.contract_id
          );
        """
   }
-
-  override def lookupMaximumLedgerTime(ids: Set[ContractId])(implicit
-      loggingContext: LoggingContext
-  ): Future[Option[Instant]] =
-    dispatcher
-      .executeSql(metrics.daml.index.db.lookupMaximumLedgerTimeDbMetrics) { implicit connection =>
-        committedContracts.lookupMaximumLedgerTime(ids)
-      }
-      .map(_.get)
-
 }
 
 private[appendonlydao] object ContractsReader {
+  private val EventKindCreated = 10
+  private val EventKindArchived = 20
+  private val contractWithoutValueRowParser: RowParser[String] =
+    str("template_id")
+  private val fullDetailsContractRowParser: RowParser[
+    (Option[String], Set[Party], Option[InputStream], Option[Int], Int, Option[Instant])
+  ] =
+    str("template_id").? ~ flatEventWitnessesColumn("flat_event_witnesses") ~ binaryStream(
+      "create_argument"
+    ).? ~ int(
+      "create_argument_compression"
+    ).? ~ int("event_kind") ~ get[Instant](
+      "ledger_effective_time"
+    )(anorm.Column.columnToInstant).? map SqlParser.flatten
+  private val contractRowParser: RowParser[(String, InputStream, Option[Int])] =
```

Comment:
Please add a blank line for faster mental parsing.
```suggestion
  
  private val contractRowParser: RowParser[(String, InputStream, Option[Int])] =
```

Reply:
Done